iocore/net/QUICNetVConnection.cc

```diff
@@ -452,12 +452,10 @@ QUICNetVConnection::start()
   QUICCCConfigQCP cc_config(this->_quic_config);
   QUICLDConfigQCP ld_config(this->_quic_config);
   this->_congestion_controller = new QUICCongestionController(this, cc_config);
-  for (auto s : QUIC_PN_SPACES) {
-    int index            = static_cast<int>(s);
-    QUICLossDetector *ld = new QUICLossDetector(this, this->_congestion_controller, &this->_rtt_measure, index, ld_config);
-    this->_frame_dispatcher->add_handler(ld);
-    this->_loss_detector[index] = ld;
-  }
+  this->_loss_detector         = new QUICLossDetector(this, this->_congestion_controller, &this->_rtt_measure, ld_config);
+  this->_congestion_controller->bind_loss_detector(this->_loss_detector);
```

Comment:
Introduced the dependency with the loss_detector. Because the congestion period needs loss detector's private vars.

Reply:
Instead of making dependency with loss_detector. It would better to hide the congestion_control as loss_detector member. and expose the RTTMeaseure's api directly with loss_detector.

Reply:
I haven’t looked into the code at all so I may not understand the reason, but creating a congestion controller inside a loss detector doesn’t sound great idea because they are independent. I’d rather have some interface instead of passing NewReno, BBR, etc to a loss detector.

Reply:
IMO, use of `QUICCCConfig` is basically the same. LD is still responsible for creating some CC. "Loss Detection" should be possible without CC.

How about passing RTTProvider (_rtt_measure) to CC?

Reply:
You mean RTTProvider has some rtt member. And ld change the RTTProvider member directly.

```
 class QUICRTTProvider
 {
 public:
   virtual ink_hrtime smoothed_rtt() const = 0;
private:
  smooth_rtt;
  rtt_val;
  ....
 };

```

Reply:
> You mean RTTProvider has some rtt member. 

I'd let `RTTMeasure` has the variables.

> And ld change the RTTProvider member directly.

Maybe we can move the code in `QUICLossDetector::_update_rtt` to RTTMeasure?

Reply:
Thanks, will change.

Reply:
done